src/control/lib/hardware/sysfs/provider.go

```diff
@@ -216,6 +248,33 @@ func (s *Provider) getPCIAddress(path string) (*hardware.PCIAddress, error) {
 	return nil, errors.Errorf("unable to parse PCI address from %q", path)
 }
 
+func (s *Provider) addNetvscDevice(topo *hardware.Topology, path string) error {
+	ifaceName := filepath.Base(path)
+
+	virt := &hardware.VirtualDevice{
+		Name: ifaceName,
+		Type: hardware.DeviceTypeNetInterface,
+	}
+
+	backingDev, err := s.getBackingDevice(path, topo.AllDevices())
+	if err != nil {
+		s.log.Noticef("Skipping NetVSC network adapter %q: not found physical backing device",
+			ifaceName)
+		return nil
+	}
+	s.log.Tracef("NetVSC network adapter %q has physical backing device %q",
+		ifaceName, backingDev.DeviceName())
+	virt.BackingDevice = backingDev
+
+	s.log.Tracef("Adding NetVSC network adapter at %q", path)
+	if len(topo.VirtualDevices) == 0 {
+		topo.VirtualDevices = make([]*hardware.VirtualDevice, 0)
+	}
+	topo.VirtualDevices = append(topo.VirtualDevices, virt)
```

Comment:
minor nit: It's not necessary to `make()` a slice before appending to it, so the length check isn't needed here. The `append()` built-in will take care of this automatically. You may be thinking of maps, where it is an error to assign to a map if it hasn't been initialized first.

https://go.dev/play/p/Apxw7h8peXa

Reply:
- [ ] Remove useless slice initialization with make